code/drasil-printers/Language/Drasil/Printing/Import/DisplayExpr.hs

```diff
@@ -36,4 +36,4 @@ dispExpr (BinOp b l r)      sm = P.Row [dispExpr l sm, P.MO $ deBinOp b, dispExp
 dispExpr (AssocBinOp b des) sm = P.Row $ intersperse (P.MO op) $ map (dispExpr' sm prec) des
   where prec = dePrecAssoc  b
         op   = deAssocBinOp b
--- dispExpr (ForAll u _ des)   sm = _        
+dispExpr (ForAll u _ des)   sm = P.Row [dispExpr des sm] 
```

Comment:
You're going to need to refine this display so that it actually shows the "forall". I think everything's already there in drasil-printers to support that.

Reply:
like dispExprForAll? or deForAll?

Reply:
It possibly doesn't need a new function. I meant displaying a proper forall, with the bound variable, etc.  There used to be something in `Import/Expr.hs` about a year ago that did that. Look through the history?

Reply:
hmm.. I can't seem to access the older commits for the git history of this file?

Reply:
Not `Import/DisplayExpr.hs`, `Import/Expr.hs`.

Reply:
The file got moved around, so it will be a bit difficult to search for it. You will need to check the original `Import.hs` file, and see from where the code came into that file, and continue the process until you found when it was removed from the file it was in.

To also give you an alternative route ... the goal is to display something that _approximately_ looks like `∀ x ∈ S • R`, where `x` is the variable, `S` is the space, and `R` is the proposition (here, it's just a DisplayExpr). The code looks like it currently only displays the `R`/proposition of the "ForAll". So, you need to add elements to the Row to also display (in order):
1. The `∀` symbol (you will need to define this one somewhere, probably the Ops datatype where the other ones are already)
2. The variable (see how similar `Expr`s are displayed -- `C` may be particularly helpful)
3. The `∈` symbol
4. The space from which the variable belongs (again, see how similar Spaces are displayed)
5. The `•` symbol
6. The proposition

There will probably be a few things further that we have to do to ensure that it is displayed properly, and there's also different styles of writing it, but I think this would be a good start.

Reply:
Ah, just saw that you posted too. It looks like comments don't appear here in reviews as they're posted! It only shows active updates in normal conversations.